client/components/deckEditor.js

```diff
@@ -9,13 +9,17 @@ export const deckEditor = {
       dynamicProperties() {
         var properties = [];
         for(const face of this.widgetState.faceTemplates) {
-          for (const object of face.objects) {
-           if(object.valueType == 'dynamic'){
-             var propObject = {name: object.value, type: object.type}
-             if (!properties.some(prop => prop.name == propObject.name))
-                properties.push(propObject);
-        }}}
-      return properties;
+          for(const object of face.objects) {
+            if(typeof object.dynamicProperties == 'object') {
+              for(const parameter in object.dynamicProperties) {
+                var propObject = {name: object.dynamicProperties[parameter], type: object.type}
```

Comment:
I think I need to change this so that `type` is `"text"` unless `parameter` is `"value"` and `object.type` is `"image"`. Otherwise the deck editor will provide an upload button for any dynamicProperties on image face objects (not just the value property).